xds/internal/client/xds.go

```diff
@@ -144,14 +144,31 @@ func unwrapHTTPFilterConfig(config *anypb.Any) (proto.Message, string, error) {
 	return s, s.GetTypeUrl(), nil
 }
 
-func validateHTTPFilterConfig(cfg *anypb.Any, lds bool) (httpfilter.Filter, httpfilter.FilterConfig, error) {
+func validateHTTPFilterConfig(cfg *anypb.Any, lds, optional, validateServer, validateClient bool) (httpfilter.Filter, httpfilter.FilterConfig, error) {
```

Comment:
Optional needs to be supported within the function, or we need to return a magic error so it can be treated specially by the caller, which I dislike (parse/unmarshal errors should *not* be ignored).  I moved the server/client type assertion to the caller though.